lib/resty/etcd/cluster/healthcheck.lua

```diff
@@ -0,0 +1,200 @@
+local healthcheck
+local require       = require
+local ipairs        = ipairs
+local pcall         = pcall
+local tostring      = tostring
+local tonumber      = tonumber
+local type          = type
+local next          = next
+local setmetatable  = setmetatable
+local getmetatable  = getmetatable
+local utils         = require("resty.etcd.utils")
+local ngx_shared    = ngx.shared
+local checker
+
+local headthcheck_endpoint = {
+    ["/v3"] = "/health",
+    ["/v3beta"] = "/health",
+}
+
+
+local fixed_field_metatable = {
+    __index =
+    function(_, k)
+        utils.log_error("field " .. tostring(k) .. " does not exist", 3)
+    end,
+    __newindex =
+    function(_, k, _)
+        utils.log_error("attempt to create new field " .. tostring(k), 3)
+    end,
+}
+
+
+local function tbl_copy(orig)
+    local orig_type = type(orig)
+    local copy
+    if orig_type == "table" then
+        copy = {}
+        for orig_key, orig_value in next, orig, nil do
+            copy[tbl_copy(orig_key)] = tbl_copy(orig_value)
+        end
+        setmetatable(copy, tbl_copy(getmetatable(orig)))
+    else -- number, string, boolean, etc
+        copy = orig
+    end
+    return copy
+end
+
+
+local function tbl_copy_merge_defaults(t1, defaults)
+    if t1 == nil then t1 = {} end
+    if defaults == nil then defaults = {} end
+    if type(t1) == "table" and type(defaults) == "table" then
+        local copy = {}
+        for t1_key, t1_value in next, t1, nil do
+            copy[tbl_copy(t1_key)] = tbl_copy_merge_defaults(
+                    t1_value, tbl_copy(defaults[t1_key])
+            )
+        end
+        for defaults_key, defaults_value in next, defaults, nil do
+            if t1[defaults_key] == nil then
+                copy[tbl_copy(defaults_key)] = tbl_copy(defaults_value)
+            end
+        end
+        return copy
+    else
+        return t1 -- not a table
+    end
+end
+
+
+local DEFAULTS = setmetatable({
+    active = {
+        concurrency = 5,
+        timeout = 5,
+        http_path = "/health",
+        host = "",
+        type = "http",
+        req_headers = {"UA: curl/7.29.0"},
```

Comment:
Actually I mean `User-Agent: lua-resty-etcd`

Reply:
I thought it was going to be abbreviated……